### ISSUE TITLE
Nesting Schema Recursively Fails when using `lambda`: "AttributeError: 'function' object has no attribute 'fields'"

### ISSUE THREAD

**Issue by hf-kklein** [fuhrysteve/marshmallow-jsonschema#164]
This works:
```python
from marshmallow import Schema, fields
from marshmallow_jsonschema import JSONSchema


class RecursiveSchema(Schema):
    # when nesting recursively you can only refer the recursive schema by its name
    nested_mwe_recursive = fields.Nested('RecursiveSchema')  # this works

json_schema = JSONSchema()
recursive_schema = json_schema.dump(RecursiveSchema())
```

But this doesn't:
```python
from marshmallow import Schema, fields
from marshmallow_jsonschema import JSONSchema


class RecursiveSchema(Schema):
    # but using the ("compile" time safe) lambda fails:
    nested_mwe_recursive = fields.Nested(lambda: RecursiveSchema()) # does not work

json_schema = JSONSchema()
recursive_schema = json_schema.dump(RecursiveSchema())
# AttributeError: 'function' object has no attribute 'fields'
```

The [marshmallow docs use the same nesting technique](https://marshmallow.readthedocs.io/en/stable/nesting.html#nesting-a-schema-within-itself) so I'd expect this to work with the jsonschema, too.

**Comment by hf-kklein** [comment]
I suggested a fix for this in PR #172

**Comment by fuhrysteve** [comment]
Thanks for the PR!
